dashboard/app/models/concerns/partial_registration.rb

```diff
@@ -47,6 +47,12 @@ def self.persist_attributes(session, user)
     session[USER_ATTRIBUTES_SESSION_KEY] = user.attributes
   end
 
+  def self.cancel(session)
+    SignUpTracking.end_sign_up_tracking(session)
+    session.delete(USER_ATTRIBUTES_SESSION_KEY)
+    session
```

Comment:
Is there a particular reason to return `session` here?

Reply:
for unit tests, which i feel kind of meh about. being able to create a fake session, pass it in to the method (which is necessary either way), and return it made testing this method more straightforward